File: NDKDemo/sample/src/main/java/com/chang/android/ndk/sample/MainActivity.java
package com.chang.android.ndk.sample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.chang.android.jna.demo.JnaHmiTest;
import com.planet.light2345.seclib.SecService;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

//        String string = SecService.encryptString(this, "xiaochang");
//        String message = "调用so加密后的结果：" + string;
//        Log.e(TAG, message);
//        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();

        // 加载三方动态库
        try {
            System.loadLibrary("sysi4hmi");
            Toast.makeText(this, "加载成功", Toast.LENGTH_SHORT).show();
            Log.e(TAG, "加载成功");
        } catch (Exception e) {
            e.printStackTrace();
            Toast.makeText(this, "加载失败", Toast.LENGTH_SHORT).show();
            Log.e(TAG, "加载失败");
        }

        new JnaHmiTest().testHmiMethod();
    }
}

File: AIDLDemo/client_server/src/main/java/com/chang/android/aidl/client_server/MainActivity.kt
package com.chang.android.aidl.client_server

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import com.chang.android.aidl.service.IDataCallback
import com.chang.android.aidl.service.Person

class MainActivity : AppCompatActivity() {

    companion object {
        private const val TAG = "MainActivity";
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    /**
     * RemoteService 在 包名+remote 进程
     */
    fun onBindRemoteServe(view: View) {
        Log.d(TAG, "开始绑定远程服务")
        ClientManager.getInstance().init(this)
    }

    fun onAddPerson(view: View) {
        val person = Person(
            "张三",
            20
        )
        ClientManager.getInstance().onAddPerson(person)
        Toast.makeText(this, "添加person $person 成功", Toast.LENGTH_SHORT).show()
    }

    fun onRemovePerson(view: View) {
        ClientManager.getInstance().removePerson(0)
    }

    fun onUpdatePerson(view: View) {
        val person = Person(
            "张三",
            22
        )
        ClientManager.getInstance().updatePerson(person)
        Toast.makeText(this, "更新person $person 成功", Toast.LENGTH_SHORT).show()
    }

    fun onQueryAllPerson(view: View) {
        val queryAll = ClientManager.getInstance().queryAll()
        Log.d(TAG, "所有person : ${queryAll?.toString()}")
        Toast.makeText(this, "所有person : ${queryAll?.toString()}", Toast.LENGTH_SHORT).show()
    }

    fun onAsyncLoadAllPerson(view: View) {
        ClientManager.getInstance().loadAll(object : IDataCallback.Stub() {
            override fun onDataReady(list: MutableList<Person>?) {
                Log.d(TAG, "异步获取所有person成功 : ${list?.toString()}")
                Toast.makeText(this@MainActivity, "异步获取所有person成功 : ${list.toString()}", Toast.LENGTH_SHORT).show()
            }

            override fun onError(code: Int, message: String?) {
                Log.d(TAG, "异步获取所有person失败 : code=$code, message=$message")
                Toast.makeText(this@MainActivity, "异步获取所有person失败 : code=$code, message=$message\"", Toast.LENGTH_SHORT).show()
            }
        })
    }

    override fun onDestroy() {
        super.onDestroy()
        ClientManager.release()
    }
}